tools/deploy_s3.py

```diff
@@ -0,0 +1,124 @@
+import gzip
+import io
+import mimetypes
+import shutil
+from pathlib import Path
+
+import boto3
+import botocore
+import typer
+
+app = typer.Typer()
+
+
+def check_s3_object_exists(s3_client, bucket: str, object_name: str):
+    try:
+        s3_client.head_object(Bucket=bucket, Key=object_name)
+        return True
+    except botocore.exceptions.ClientError as e:
+        if e.response["Error"]["Code"] == "404":
+            return False
+
+        raise
+
+
+def _rm_s3_prefix(bucket: str, prefix: str):
+    """Remove all objects under a given prefix"""
+    s3 = boto3.resource("s3")
+    bucket = s3.Bucket(bucket)
+    for obj in bucket.objects.filter(Prefix=prefix):
+        obj.delete()
+
+
+@app.command()
+def deploy_to_s3_main(
+    local_folder: Path = typer.Argument(..., help="Path to the local folder"),
+    remote_prefix: Path = typer.Argument(..., help="Remote prefix"),
+    bucket: str = typer.Option(..., help="bucket name"),
+    cache_control: str = typer.Option(
+        "max-age=30758400, immutable, public", help="Cache control header to set"
+    ),
+    pretend: bool = typer.Option(False, help="Don't actually upload anything"),
+    overwrite: bool = typer.Option(False, help="Overwrite existing files"),
+    rm_remote_prefix: bool = typer.Option(
+        False, help="Remove existing files under the remote prefix"
+    ),
+):
+    """Deploy a dist folder with Pyodide packages to AWS S3"""
+    s3_client = boto3.client("s3")
+
+    typer.echo(f"Deploying {local_folder} to s3://{bucket}/{remote_prefix}")
+    typer.echo("Options: ")
+    typer.echo(f" - {cache_control=}")
+    typer.echo(f" - {pretend=}")
+    typer.echo(" - content-encoding: gzip")
+
+    if rm_remote_prefix:
+        _rm_s3_prefix(bucket, str(remote_prefix).lstrip("/"))
```

Comment:
Maybe we want to validate the prefix in some way to reduce chance of accidentally deleting everything?

Reply:
Shouldn't pretend also apply to deleting stuff?

Reply:
> Shouldn't pretend also apply to deleting stuff?

Yes, good that someone is reviewing this code ^^

> Maybe we want to validate the prefix in some way to reduce chance of accidentally deleting everything?

Indeed. Let's do more validation of both of the prefix in general and only allow deletion of prefixes ending in "/dev/".